addons/mail/static/src/core/common/thread_model.js

```diff
@@ -672,17 +668,14 @@ export class Thread extends Record {
             return;
         }
         await this.store.chatHub.initPromise;
-        const cw = this.store.ChatWindow.insert(
-            assignDefined({ thread: this }, { fromMessagingMenu, bypassCompact })
-        );
-        cw.open({ focus, swapOpened });
-        return cw;
+        this.channel.chatWindow = { fromMessagingMenu, bypassCompact };
+        this.channel.chatWindow.open({ focus, swapOpened });
+        return this.channel.chatWindow;
```

Comment:
I'm not sure we gain to use the field instead of insert in this case if we have to read the field 3 times in a row.